src/cmd/build.rs

```diff
@@ -298,8 +295,108 @@ fn exec_cargo_for_wasm_target(
             .using_temp(cargo_build)?;
     }
 
-    // clear RUSTFLAGS
-    std::env::remove_var("RUSTFLAGS");
+    Ok(())
+}
+
+/// Executes `cargo dylint` with the ink! linting driver that is built during
+/// the `build.rs`.
+///
+/// We create a temporary folder, extract the linting driver there and run
+/// `cargo dylint` with it.
+fn exec_cargo_dylint(crate_metadata: &CrateMetadata, verbosity: Verbosity) -> Result<()> {
+    check_dylint_requirements(crate_metadata.manifest_path.directory())?;
+
+    let tmp_dir = tempfile::Builder::new()
+        .prefix("cargo-contract-dylint_")
+        .tempdir()?;
+    log::debug!("Using temp workspace at '{}'", tmp_dir.path().display());
+
+    let template = include_bytes!(concat!(env!("OUT_DIR"), "/ink-dylint-driver.zip"));
+    crate::util::unzip(template, tmp_dir.path().to_path_buf(), None)?;
+
+    let manifest_path = crate_metadata.manifest_path.cargo_arg()?;
+    let args = vec!["ink_linting", &manifest_path];
+    let tmp_dir_path = tmp_dir.path().as_os_str().to_string_lossy();
+    let env = vec![
+        ("DYLINT_LIBRARY_PATH", Some(tmp_dir_path.as_ref())),
+        ("DYLINT_DRIVER_PATH", Some(tmp_dir_path.as_ref())),
```

Comment:
When I remove this line the driver is cached and doesn't need to be rebuild on every check. Makes sense because you are compiling it to a new temporary directory each time.

I don't know where it is put when I remove this line, though.